goagen/codegen/types.go

```diff
@@ -115,6 +115,11 @@ func goTypeDefObject(obj design.Object, def *design.AttributeDefinition, tabs in
 			typedef = "*" + typedef
 		}
 		fname := GoifyAtt(field, name, true)
+		if tname, ok := field.Metadata["struct:field:type"]; ok {
```

Comment:
Might make sense to move the test so that the type name doesn't have to be computed if the metadata is present. So move the test to line 113, declare the variable `typedef` above and have a `if typedef == ""` before the computation done on lines 113 to 116. Wow lots of words for something that is so simple :)